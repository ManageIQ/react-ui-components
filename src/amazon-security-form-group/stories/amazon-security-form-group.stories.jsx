import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withInfo } from '@storybook/addon-info';

import AmazonSecurityFormGroup from '../amazon-security-form-group';

const vpcIds = [
  {
    value: 0,
    label: 'Davidson',
  },
  {
    value: 1,
    label: 'Coleen',
  },
  {
    value: 2,
    label: 'Matthews',
  },
  {
    value: 3,
    label: 'Andrews',
  },
  {
    value: 4,
    label: 'Ruthie',
  },
  {
    value: 5,
    label: 'Lydia',
  },
  {
    value: 6,
    label: 'Jillian',
  },
  {
    value: 7,
    label: 'Sarah',
  },
  {
    value: 8,
    label: 'Mitchell',
  },
  {
    value: 9,
    label: 'Daniels',
  },
  {
    value: 10,
    label: 'Chan',
  },
  {
    value: 11,
    label: 'Cooper',
  },
  {
    value: 12,
    label: 'Stewart',
  },
  {
    value: 13,
    label: 'Patrice',
  },
  {
    value: 14,
    label: 'Cheryl',
  },
];

const simulateRequest = () => new Promise((resolve) => {
  setTimeout(() => resolve(vpcIds), 1500);
});

storiesOf('Amazon Security forms', module).add('Amazon Security form group', withInfo()(() => (
  <AmazonSecurityFormGroup
    onSave={action('onSubmit')}
    onCancel={action('onCancel')}
    loadData={simulateRequest}
    updateFormState={values => console.log(values)}
  />
)));
