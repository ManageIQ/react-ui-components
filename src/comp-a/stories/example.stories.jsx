import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import '../styles.scss';
import { ExampleCmp } from '../some-cmp/example.component';
import Provider from './provider';

storiesOf('Example', module)
  .add('Stateless component', () => <ExampleCmp onChangeText={action('change text')} onChangeType={action('change type')} />)
  .addDecorator(story => <Provider story={story()} />)
  .add('Statefull component', () => <div />);
