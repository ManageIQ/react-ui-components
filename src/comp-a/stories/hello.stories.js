import React from 'react';

import { storiesOf } from '@storybook/react';

import '../styles.scss';

import { HelloCmp } from '../some-cmp/hello.component';

storiesOf('Hello', module)
  .add('Hello example', () => <HelloCmp />);
