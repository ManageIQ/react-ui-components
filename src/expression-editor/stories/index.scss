@import '~react-bootstrap-typeahead/css/Typeahead.css';
@import "../sass/style.scss";
