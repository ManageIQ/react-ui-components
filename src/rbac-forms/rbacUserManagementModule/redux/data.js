export const data = {
  rows: [
    {
      id: 100,
      name: 'Adam the first',
      userid: 'Admin',
      email: 'email@neco.com',
      current_group: {
        label: 'Some group',
        onClick: () => console.log('Current group clicked'),
      },
      role: {
        label: 'Admin role',
        onClick: () => console.log('Current role click'),
      },
      groups: [{
        label: 'Cloud-Operators',
        icon: 'group',
        groupId: 1,
        onClick: () => console.log('Group click'),
      }, {
        label: 'Cloud-Users',
        icon: 'group',
        groupId: 2,
        onClick: () => console.log('Group click'),
      }],
      lastlogon: '06/25/18 06:58:14 UTC',
      lastlogoff: '06/30/18 06:58:14 UTC',
    },
    {
      id: 101,
      name: 'Marge the middle',
      userid: 'La transparenza della qualita',
      email: 'email@jandak.com',
      current_group: {
        label: 'Mediocre group',
        onClick: () => console.log('Current group clicked'),
      },
      role: {
        label: 'Shot role',
        onClick: () => console.log('Current role click'),
      },
      groups: [{
        label: 'Cloud-Operators',
        icon: 'group',
        groupId: 1,
        onClick: () => console.log('Group click'),
      }, {
        label: 'Cloud-Users',
        icon: 'group',
        groupId: 2,
        onClick: () => console.log('Group click'),
      }],
      lastlogon: '01/25/18 06:58:14 UTC',
      lastlogoff: '06/30/18 06:58:14 UTC',
    },
    {
      id: 102,
      name: 'Zlatko the last',
      userid: 'Peligro',
      email: 'email@jandak.com',
      current_group: {
        label: 'Admin group',
        onClick: () => console.log('Current group clicked'),
      },
      role: {
        label: 'Cloud-Users',
        onClick: () => console.log('Current role click'),
      },
      groups: [{
        label: 'Cloud-Operators',
        icon: 'group',
        groupId: 1,
        onClick: () => console.log('Group click'),
      }, {
        label: 'Cloud-Users',
        icon: 'group',
        groupId: 2,
        onClick: () => console.log('Group click'),
      }],
      lastlogon: '08/25/18 06:58:14 UTC',
      lastlogoff: '010/30/18 06:58:14 UTC',
    },
  ],
  columns: [{
    property: 'name',
    label: 'Full Name',
  }, {
    property: 'userid',
    label: 'Username',
  }, {
    property: 'email',
    label: 'E-mail',
  }, {
    property: 'current_group',
    label: 'Current Group',
  }, {
    property: 'role',
    label: 'Role',
  }, {
    property: 'lastlogon',
    label: 'Last Logon',
  }, {
    property: 'lastlogoff',
    label: 'Last Logoff',
  }],
  selectedUsers: null,
  groups: [
    {
      value: 1,
      label: 'Hope',
    },
    {
      value: 2,
      label: 'Christa',
    },
    {
      value: 3,
      label: 'Tania',
    },
    {
      value: 4,
      label: 'Mcintyre',
    },
    {
      value: 5,
      label: 'Tonya',
    },
    {
      value: 6,
      label: 'Marcie',
    },
    {
      value: 7,
      label: 'Cara',
    },
    {
      value: 8,
      label: 'Becker',
    },
    {
      value: 9,
      label: 'Foley',
    },
    {
      value: 10,
      label: 'Perkins',
    },
  ],
};
