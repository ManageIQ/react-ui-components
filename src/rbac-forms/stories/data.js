import { action } from '@storybook/addon-actions';

export const groups = [
  {
    value: 1,
    label: 'Hope',
  },
  {
    value: 2,
    label: 'Christa',
  },
  {
    value: 3,
    label: 'Tania',
  },
  {
    value: 4,
    label: 'Mcintyre',
  },
  {
    value: 5,
    label: 'Tonya',
  },
  {
    value: 6,
    label: 'Marcie',
  },
  {
    value: 7,
    label: 'Cara',
  },
  {
    value: 8,
    label: 'Becker',
  },
  {
    value: 9,
    label: 'Foley',
  },
  {
    value: 10,
    label: 'Perkins',
  },
];

export const editedUser = {
  name: 'Users full name',
  userid: 'admin jiny',
  chosen_group: [5, 7],
};

export const usersTableColumns = [{
  property: 'fullname',
  label: 'Full Name',
}, {
  property: 'username',
  label: 'Username',
}, {
  property: 'email',
  label: 'E-mail',
}, {
  property: 'currentgroup',
  label: 'Current Group',
}, {
  property: 'role',
  label: 'Role',
}, {
  property: 'lastlogon',
  label: 'Last Logon',
}, {
  property: 'lastlogoff',
  label: 'Last Logoff',
}];

export const usersTableRows = [
  {
    id: 100,
    fullname: 'Adam the first',
    username: 'Admin',
    email: 'email@neco.com',
    currentgroup: 'Some group',
    role: 'Admin role',
    lastlogon: '06/25/18 06:58:14 UTC',
    lastlogoff: '06/30/18 06:58:14 UTC',
  },
  {
    id: 101,
    fullname: 'Marge the middle',
    username: 'La transparenza della qualita',
    email: 'email@jandak.com',
    currentgroup: 'Mediocre group',
    role: 'Shot role',
    lastlogon: '01/25/18 06:58:14 UTC',
    lastlogoff: '06/30/18 06:58:14 UTC',
  },
  {
    id: 102,
    fullname: 'Zlatko the last',
    username: 'Peligro',
    email: 'email@jandak.com',
    currentgroup: 'Super group',
    role: 'Shot role',
    lastlogon: '08/25/18 06:58:14 UTC',
    lastlogoff: '010/30/18 06:58:14 UTC',
  },
];

export const user = {
  name: 'Administrator',
  userid: 'Admin',
  email: 'mail@mail.com',
  current_group: {
    label: 'EvmGroup-super_administrator',
    onClick: () => action('Current group clicked'),
  },
  groups: [{
    label: 'Cloud-Operators',
    icon: 'group',
    groupId: 1,
    onClick: () => action('Group click'),
  }, {
    label: 'Cloud-Users',
    icon: 'group',
    groupId: 2,
    onClick: () => action('Group click'),
  }],
  role: {
    label: 'Cloud-Users',
    onClick: () => action('Current role click'),
  },
};
