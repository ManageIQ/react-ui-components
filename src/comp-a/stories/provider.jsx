import React from 'react';
import { createStore } from 'redux';
import PropTypes from 'prop-types';
import ExampleConnected from '../redux/example';
import exampleApp from '../reducers';

const store = createStore(exampleApp);

if (module.hot) {
  module.hot.accept('../reducers', () => {
    const nextRootReducer = exampleApp;
    store.replaceReducer(nextRootReducer);
  });
}

export default function Provider({ story }) {
  return (
    <ExampleConnected store={store}>
      {story}
    </ExampleConnected>
  );
}

Provider.propTypes = {
  story: PropTypes.any,
};

Provider.defaultProps = {
  story: '',
};
