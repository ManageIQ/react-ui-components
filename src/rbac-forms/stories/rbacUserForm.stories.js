import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withInfo } from '@storybook/addon-info';
import { RbacUserForm, RbacUserPreview } from '../';
import { GenericPreviewTable } from '../../table/';
import { groups, usersTableColumns, usersTableRows, user, editedUser } from './data';

storiesOf('Rbac forms', module).add('Rbac add user form', withInfo()(() => (
  <RbacUserForm
    groups={groups}
    onSave={action('onSave')}
    onCancel={action('onCancel')}
  />
))).add('Rbac edit user form', withInfo()(() => (
  <RbacUserForm
    groups={groups}
    onSave={action('onSave')}
    onCancel={action('onCancel')}
    initialValues={editedUser}
    editDisabled={false}
  />
))).add('Rbac user preview', withInfo()(() => (
  <RbacUserPreview user={user} />
))).add('Rbac users table', withInfo()(() => (
  <GenericPreviewTable
    rows={usersTableRows}
    columns={usersTableColumns}
    rowClick={action('Row clicked')}
    rowSelect={action('User selected')}
    showIcon
    showSelect
    icon={{
      type: 'pf',
      name: 'user',
    }}
  />
)));
