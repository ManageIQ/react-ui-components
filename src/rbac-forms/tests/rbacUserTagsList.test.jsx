import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { RbacUserTagsList } from '../';
import { tags } from '../stories/data';

describe('Rbac user tags list component', () => {
  it('should render conrrectly with tags', () => {
    const wrapper = shallow(<RbacUserTagsList tags={tags} tenant="Foo" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render conrrectly without tags', () => {
    const wrapper = shallow(<RbacUserTagsList tenant="Foo" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
