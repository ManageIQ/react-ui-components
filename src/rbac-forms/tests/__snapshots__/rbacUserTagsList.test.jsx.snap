// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rbac user tags list component should render conrrectly with tags 1`] = `
<div>
  <h3>
    Smart Management
  </h3>
  <div
    className="form-horizontal static"
  >
    <div
      className="form-group"
    >
      <label
        className="col-md-2 control-label"
      >
         
        %s Tags
      </label>
      <div
        className="col-md-8"
      >
        <Icon
          name="tag"
          type="fa"
        />
         
        Tag 1: Tag value 1
        <br />
        <Icon
          name="tag"
          type="fa"
        />
         
        Tag 2: Tag value 2
        <br />
        <Icon
          name="tag"
          type="fa"
        />
         
        Tag 3: Tag value 3
        <br />
        <Icon
          name="tag"
          type="fa"
        />
         
        Tag 4: Tag value 4
        <br />
        <Icon
          name="tag"
          type="fa"
        />
         
        Tag 5: Tag value 5
        <br />
      </div>
    </div>
  </div>
</div>
`;

exports[`Rbac user tags list component should render conrrectly without tags 1`] = `
<div>
  <h3>
    Smart Management
  </h3>
  <div
    className="form-horizontal static"
  >
    <div
      className="form-group"
    >
      <label
        className="col-md-2 control-label"
      >
         
        %s Tags
      </label>
      <div
        className="col-md-8"
      >
        <span>
          <Icon
            name="tag"
            type="fa"
          />
           
          No %s Tags have been assigned
        </span>
      </div>
    </div>
  </div>
</div>
`;
